src/python/pants/backend/python/target_types.py

```diff
@@ -264,15 +267,39 @@ class PythonTestsDependencies(Dependencies):
     supports_transitive_excludes = True
 
 
-class PythonRuntimeBinaryDependencies(StringSequenceField):
-    """Addresses to binary targets that will be built and included in this target at runtime.
+class PythonRuntimeBuildDependencies(StringSequenceField):
+    """Addresses to targets that can be built with the `./pants build` goal and whose resulting
+    assets should be included in the test run.
 
-    These binary targets do not necessarily need to be `python_binary` targets; they can be any
-    targets accepted by `./pants binary`
+    Pants will build the asset as if it had run `./pants build`, and will include the result in
```

Comment:
I think my issue is just with the word "asset". I understand we are trying to be general, so I think either now, or later after some stabilization, it may be good to have a canonical description of what an "asset" is here (since we expect that will expand).